### PR TITLE
chore(actions): add test GitHub workflow

### PR DESCRIPTION
**Pull request by JamesSingleton**
This PR adds the following workflows:

`tests.yml`
* This runs jest, git history test and lockfile lint test on PRs to master

`release.yml`
* This runs `semantic-release` on a push to master where the event is a pull request is merged and the base ref is master

`master-health-check.yml`
* This runs on a cron job every single day at 0800 and 1600 and run `npm run test`

### CODE REVIEW

**Comment by JamesSingleton** [comment]
https://github.com/americanexpress/iguazu/pull/23 is an example of this working from a fork

**Comment by infoxicator** [comment]
@JamesSingleton why do we need a master health check cron job?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Install dependencies
+        run: npm ci
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (nellyk) [inline .github/workflows/release.yml:25]: can you add this env variables 

```        GIT_AUTHOR_EMAIL: ${{ secrets.GIT_AUTHOR_EMAIL }}
        GIT_AUTHOR_NAME: ${{ secrets.GIT_AUTHOR_NAME }}
        GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMITTER_EMAIL }}
        GIT_COMMITTER_NAME: ${{ secrets.GIT_COMMITTER_NAME }}

**Comment by JamesSingleton** [comment]
> @JamesSingleton why do we need a master health check cron job?

@infoxicator and @nellyk, the reason I decided to have a master health check cron job was because I removed running the tests on push to master and it only runs on a PR. In my opinion, I don't think it is efficient to run tests on code that just passed the tests on a PR. Furthermore, since it runs on 10.x, 12.x and 14.x it would be able to signal issues like Node 12.17.0 removing the need to flag for esm.

File: .github/workflows/release.yml (hunk 1 above)
Comment (JamesSingleton) [reply .github/workflows/release.yml:25]: Yes I can add those, what are those exactly needed for?

**Comment by JamesSingleton** [comment]
> Are we deleting `.travis.yml` in a separate PR?

We can or do it in this one. I know @nellyk wanted me to test this out, but I had questions on how exactly to do this? It is a similar setup that I have in personal repos as well as seen among other repos using GitHub actions.

**Comment by nellyk** [comment]
@JamesSingleton can you look at getting this in too https://github.com/americanexpress/one-service-worker/pull/19. We should remove the travis.yml file as that would conflict with github actions.

**Comment by nellyk** [comment]
We would also need to fix this https://github.com/americanexpress/one-service-worker/runs/766967664?check_suite_focus=true to ensure that release commits do not trigger releases

**Comment by mtomcal** [comment]
How did you test the semantic release bit?

**Comment by JamesSingleton** [comment]
So I modeled it after https://github.com/JamesSingleton/one-app-page-wrapper/blob/master/.github/workflows/release.yml. A little difficult to test semantic-release process without actually releasing.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged && github.base_ref == 'master'
+    steps:
```
Comment (nellyk) [inline .github/workflows/release.yml:13]: can you please add this https://github.com/americanexpress/one-service-worker/pull/19

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged && github.base_ref == 'master'
```
Comment (nellyk) [inline .github/workflows/release.yml:12]: also this ```   if: "! contains(github.event.head_commit.message, '[skip ci]')"``` which i got from @JAdshead PR https://github.com/americanexpress/fetch-enhancers/pull/6

File: .github/workflows/release.yml (hunk 2 above)
Comment (nellyk) [reply .github/workflows/release.yml:13]:
```      with:
        persist-credentials: false```

File: .github/workflows/master-health-check.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: Master Health Check
+
+on:
+  schedule:
+    # At minute 0 past hour 0800 and 1600.
+    - cron:  '0 8,16 * * *'
```
Comment (10xLaCroixDrinker) [inline .github/workflows/master-health-check.yml:6]: Why 1600? I'd expect 2000

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - name: Setup Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node }}
+    - name: Install Dependencies
+      run: npm ci
+      env:
+        NODE_ENV: development
+    - name: Unit Tests
+      run: ./node_modules/.bin/jest
```
Comment (10xLaCroixDrinker) [inline .github/workflows/tests.yml:29]: How do we keep all of these in sync with what is in `npm test`?

File: .github/workflows/master-health-check.yml (hunk 4 above)
Comment (JamesSingleton) [reply .github/workflows/master-health-check.yml:6]: 1600 as in 4pm. Do you want it to run at 0800 and 2000 (8pm)? I was just doing it 8 hours apart but we can do 12

File: .github/workflows/tests.yml (hunk 5 above)
Comment (10xLaCroixDrinker) [reply .github/workflows/tests.yml:29]: @JamesSingleton and I discussed over slack and decided it is fine to take this considering how infrequently test scripts are added, and the point that any opener or reviewer of a new test script would be particularly mindful of the tests that are being run

File: .github/workflows/master-health-check.yml (hunk 4 above)
Comment (10xLaCroixDrinker) [reply .github/workflows/master-health-check.yml]: 12 hrs apart if we're running it 2x a day. I'd only do 8hrs apart if we were going 3x a day

**Pull request merged**